features/keychain/module/keychain.js

```diff
@@ -25,7 +25,7 @@ export const MODULE_ID = 'keychain'
 export class Keychain {
   #masters = Object.create(null)
   #legacyPrivToPub = null
-  #privateKeysAreLocked = false
+  #seedLockStatus = Object.create(null)
```

Comment:
easier this way than to have it in `#masters`?

Reply:
I was wondering the same, but in all honesty, on a second though, I think in the future we may want to expose the lock status per seed phrase, and then we'd have to export a subset of `#masters` which is scary. Now it's very clear in the code what `#masters` does and who interacts with it at the risk of potentially de-syncing the state.

Reply:
Yeah, honestly I did not went that forward in my thinking, thanks @kewde 😅 , but from a general point of view, I think these represent different things. e.g.:

- `masters` represent the seeds and keys
- `seedLockStatus` is more a metadata representation that is related to seeds/keys but is not the thing itself.

I think it is usually a good approach to have this type of separation of concerns in the code to avoid too much coupling

Reply:
i was thinking in the future we would actually delete private keys in `#masters`, and in that sense keeping them together in `#masters` made sense but sounds good for now